src/components/scenes/EditTokenScene.tsx

```diff
@@ -98,13 +98,37 @@ function EditTokenSceneComponent(props: Props) {
     }
 
     if (tokenId != null) {
-      await wallet.currencyConfig.changeCustomToken(tokenId, token)
+      await wallet.currencyConfig.changeCustomToken(tokenId, customTokenInput)
+      navigation.goBack()
     } else {
-      const tokenId = await wallet.currencyConfig.addCustomToken(token)
-      await wallet.changeEnabledTokenIds([...wallet.enabledTokenIds, tokenId])
-      logActivity(`Add Custom Token: ${account.username} -- ${getWalletName(wallet)} -- ${wallet.type} -- ${tokenId} -- ${currencyCode} -- ${decimals}`)
+      const { currencyConfig } = wallet
+      const builtinTokens = Object.values(currencyConfig.builtinTokens)
+
+      // Check if custom token input conflicts with built-in tokens
+      const matchingContractToken = builtinTokens.find(
+        builtInToken => builtInToken.networkLocation?.contractAddress === customTokenInput.networkLocation?.contractAddress
```

Comment:
Do not access `networkLocation`, since chains like EOS, Tron, or Ripple have wildly different stuff in there. I do see that we hard-code this field name a few lines above, which is unfortunate. I will create a task to make this scene dynamic, based on the `EdgeCurrencyInfo` data structures, so we can correctly handle custom tokens on these weird chains.

The only correct way to compare two coins is by `tokenId`, and the only way to get the `tokenId` is to call `addCustomToken`. I would love to expose an `validateCustomToken` method that calculates tokenId without adding the token, but this is may be impossible without a breaking change to the currency plugins.

So, given these constraints, it might make sense to add the token first and then issue the warning. If the user denies the warning, then delete the token. Otherwise, we need to put a big fat TODO comment in this PR, so we know the networkLocation logic is just a placeholder, pending a `validateCustomToken` method getting added.

Instead, we rely on the currency plugins to convert `networkLocation => tokenId`, and then we can use the tokenId to compare identity. Unfortunately, there is no easy way to access this conversion, aside from calling `addCustomToken`.